File: Java examples/src/builderPattern/ConcreteBuilder.java
package builderPattern;

class ConcreteBuilder implements Builder {
    private Product product = new Product();

    @Override
    public void buildPartA() {
        product.setPartA("PartA built");
    }

    @Override
    public void buildPartB() {
        product.setPartB("PartB built");
    }

    @Override
    public Product getResult() {
        return product;
    }
}

File: C# examples/BuilderPattern/ConcreteBuilder.cs
namespace BuilderPattern
{
    public class ConcreteBuilder : IBuilder
    {
        private Product _product = new Product();

        public void BuildPartA()
        {
            _product.PartA = "PartA built";
        }

        public void BuildPartB()
        {
            _product.PartB = "PartB built";
        }

        public Product GetResult()
        {
            return _product;
        }
    }
}